djvm/src/main/java/sandbox/java/util/AbstractMap.java

```diff
@@ -0,0 +1,78 @@
+package sandbox.java.util;
+
+import java.io.Serializable;
+
+/**
+ * This is a dummy class that implements just enough of {@link java.util.AbstractMap}
+ * to allow us to compile {@link sandbox.java.util.LinkedHashMap} and
+ * {@link sandbox.java.util.concurrent.ConcurrentHashMap}.
+ */
+public abstract class AbstractMap<K, V> extends sandbox.java.lang.Object implements Map<K, V> {
+
+    protected AbstractMap() {
+    }
+
+    @Override
+    public abstract Set<Map.Entry<K,V>> entrySet();
+
+    @Override
+    public Collection<V> values() {
+        throw new UnsupportedOperationException("Dummy class - not implemented");
+    }
+
+    @Override
+    public Set<K> keySet() {
+        Set<Entry<K, V>> entries = entrySet();
+        HashSet<K> keys = new HashSet<>(entries.size());
+        for (Entry<K, V> entry : entries) {
+            keys.add(entry.getKey());
+        }
+        return keys;
```

Comment:
Wouldnt just `TreeSet` be enough?

Reply:
It's a dummy class that isn't used inside the sandbox.

Reply:
As a reader, I still find it unclear to what extent this is "dummy", I see that `values()` is unimplemented but `keySet()` is, why?

Reply:
I was expecting to need `keySet()`, and its implementation was trivial enough. But I knew I wouldn't need `values()`.